tests/http/test_request_normalization.py

```diff
@@ -205,3 +205,19 @@ def test_werkzeug_request_normalization(httpbin):
     assert request.url == httpbin.url + '/put?key=val'
     assert request.method == 'put'
     assert request.data == {'key2': 'val2'}
+
+
+@pytest.mark.skipif(not _aiohttp_available, reason="aiohttp not installed")
+def test_aiohttp_request_normalization():
+    from aiohttp.test_utils import make_mocked_request
+
+    raw_request = make_mocked_request(
```

Comment:
I'm not opposed to this since it is from the formal `aiohttp` tools, however, it is pretty easy to do async testing with pytest.

```python
@pytest.mark.asyncio
async def my_async_test():
    response = await ....
```

Since we have the `httpbin` fixture which runs a local webserver it should be pretty easy to generate a real request/response.

Reply:
Thank you for the feedback. I would also have to use the compatibility utils from asyncio to not get a syntax error in Python < 3.5.

I may try this if I find the time, since the `aiohttp.test_utils` API is marked as provisional, which may break this test in the future.

Reply:
So this is tricky, since the tests need to pass for Python 2.

An `async` keyword just triggers a syntax error, with no chance for pytest to even skip the test. Even if I use the Python <3.5 `@asyncio.coroutine` syntax, I get a syntax error due to the `yield from` statement.

It looks to me like the only way to get actual async tests to run for newer versions is to move them to their own file, and ignore them during test collection, e.g. like this: https://github.com/pytest-dev/pytest/issues/2817 (which may have further side effects; after all, we don't want to skip any other test files on syntax errors).

@pipermerriam does this still seem worth exploring to you?